eng/pipelines/templates/steps/use-node-test-version.yml

```diff
@@ -62,3 +62,4 @@ steps:
       }
 
     displayName: Reinstall native dependencies
+    condition: ne(variables['NodeTestVersion'], variables['NodeVersion'])
```

Comment:
We probably also want to add `succeeded()` to this condition so we don't do this work in a failed or canceled build.